### PR TITLE
Remove the `With<Parent>` query filter from `bevy_ui::render::extract_uinode_borders`

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

Remove the `With<Parent>` query filter from the `parent_node_query` parameter of the `bevy_ui::render::extract_uinode_borders` function. This is a bug, the query is only used to retrieve the size of the current node's parent. We don't care if that parent node has a `Parent` or not.

### CODE REVIEW

**Review by nicopap — APPROVED** [review]

**Review by Selene-Amanita — APPROVED** [review]
Looks good to me, but I have a naive question: when do we check the coherence of a hierarchy and when do we not do it? `propagate_transforms` does it for example, but maybe it's only to prevent an infinite recursion?

File: crates/bevy_ui/src/render/mod.rs [hunk 1]
```diff
@@ -277,7 +277,7 @@ pub fn extract_uinode_borders(
             Without<ContentSize>,
         >,
     >,
-    parent_node_query: Extract<Query<&Node, With<Parent>>>,
+    parent_node_query: Extract<Query<&Node>>,
```
Comment (mockersf) [inline crates/bevy_ui/src/render/mod.rs:280]:
```suggestion
    node_query: Extract<Query<&Node>>,
```

**Comment by james7132** [comment]
> `propagate_transforms` does it for example, but maybe it's only to prevent an infinite recursion?

It does it because it would be unsound use of unsafe if the hierarchy is improperly structured, due to the mutable access on multiple threads.

File: crates/bevy_ui/src/render/mod.rs (hunk 1 above)
Comment (nicopap) [reply crates/bevy_ui/src/render/mod.rs:280]: Well, since it's the query used to get the parent, it makes sense to be named "parent_node_query"

Comment (mockersf) [reply crates/bevy_ui/src/render/mod.rs:280]: I would prefer to name it based on what is is, not where it's used

Comment (ickshonpe) [reply crates/bevy_ui/src/render/mod.rs]: > I would prefer to name it based on what is is, not where it's used

agreed I think, but keep the name `parent_node` for the value retrieved from the query, where it is the actual Node component of the parent.

**Comment by ickshonpe** [comment]
> Looks good to me, but I have a naive question: when do we check the coherence of a hierarchy and when do we not do it? `propagate_transforms` does it for example, but maybe it's only to prevent an infinite recursion?

There are also checks in `UiSurface::get_layout` and `UiSurface::update_children` but they probably aren't sufficient for identifying all the possible problems. I think we still just panic if a node in the middle of the UI layout tree is removed without removing its descendants as well.

**Comment by alice-i-cecile** [comment]
@ickshonpe I think the rename was incomplete: CI failures look real. Once CI is passing this will be merged!

**Comment by ickshonpe** [comment]
Ah yep I see, I'll fix it now.

**Review by mockersf — APPROVED** [review]

**Pull request merged**